trade-finance-manager-ui/server/controllers/utilisation-reports/record-corrections/create-record-correction-request/form-helpers.ts

```diff
@@ -0,0 +1,50 @@
+import { RECORD_CORRECTION_REASON, RecordCorrectionReason } from '@ukef/dtfs2-common';
+import { CreateRecordCorrectionRequestFormValues } from '../../../../types/view-models';
+
+/**
+ * Values for the record correction reasons as a string array.
+ */
+const RECORD_CORRECTION_REASON_VALUES = Object.values(RECORD_CORRECTION_REASON) as string[];
+
+export type CreateRecordCorrectionRequestFormRequestBody = {
+  reasons?: string | string[];
+  additionalInfo?: string;
+};
+
+/**
+ * Checks if the provided reason is a valid RecordCorrectionRequestReason.
+ * @param reason - The reason to validate.
+ * @returns True if the reason is valid, otherwise false.
+ */
+export function isRecordCorrectionReason(reason: string): reason is RecordCorrectionReason {
+  return RECORD_CORRECTION_REASON_VALUES.includes(reason);
+}
+
+/**
+ * Retrieves valid record correction reasons from the provided input.
+ * @param reasons - The reasons to validate. Can be a single reason as a string or an array of reasons.
+ * @returns An array of valid record correction reasons. If no valid reasons are found, returns an empty array.
+ */
+export const extractRecordCorrectionReasons = (reasons?: string | string[]): RecordCorrectionReason[] => {
+  if (!reasons) {
+    return [];
+  }
+
+  if (Array.isArray(reasons)) {
+    return reasons.filter((reason) => isRecordCorrectionReason(reason));
+  }
+
+  return isRecordCorrectionReason(reasons) ? [reasons] : [];
+};
+
+/**
+ * Extracts form values from the request body for creating a record correction request.
+ * @param requestBody - The request body containing form data.
+ * @returns Object containing the extracted form values.
+ */
+export const extractCreateRecordCorrectionRequestFormValues = (
```

Comment:
What do you think about simplifying the namings here to e.g

```
extractFormValues
extractReasons
```

Since we're already in this directory
```
utilisation-reports/record-corrections/create-record-correction-request
```

Reply:
I feel a bit against this as they're exported and I think it's quite nice that the context "create record correction request" is included in the function name so we don't have to rely on looking at the import file path